packages/perseus/src/widgets/interactive-graphs/graphs/circle.tsx

```diff
@@ -233,6 +238,50 @@ function getCircleGraphDescription(
     return strings.srCircleInteractiveElement;
 }
 
+export const getCircleKeyboardConstraint = (
+    center: vec.Vector2,
+    radiusPoint: vec.Vector2,
+    snapStep: vec.Vector2,
+): {
+    up: vec.Vector2;
+    down: vec.Vector2;
+    left: vec.Vector2;
+    right: vec.Vector2;
+} => {
+    // Create a helper function that moves the point and then checks
+    // if it overlaps with the center point after the move.
+    const movePointWithConstraint = (
```

Comment:
This function is recreated in a few places, albeit with several differences to support each graph type. I ruminated on whether to move this to a util file, but I think the subtle differences and value of having this logic "in situ" outweigh what we would gain by keeping this more DRY. 

I'm open to other interpretations on this, however!

Reply:
I could go either way between your current implementation and a util function that you pass in the type to so you can easily see what the differences are. Since you've already written it this way, I say leave it as is.